.github/workflows/automerge_staging.yml

```diff
@@ -0,0 +1,85 @@
+name: AutoMerge (staging)
+on:
+  pull_request:
+
+# https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#permissions
+permissions: read-all
+
+jobs:
+  AutoMerge_staging:
+    timeout-minutes: 60
+    if: github.event_name != 'pull_request' || github.repository == github.event.pull_request.head.repo.full_name
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        version:
+          - '1'
+        os:
+          - ubuntu-latest
+        arch:
+          - x64
+    steps:
+      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
+      - uses: julia-actions/setup-julia@0b9b1d2cd24245f151902702d8e73b3f6b910014 # v1.6.0
+        with:
+          version: ${{ matrix.version }}
+      - run: println("::set-output name=manifest_version::$(VERSION.major).$(VERSION.minor)")
+        shell: julia --color=yes --project=.ci/ {0}
+        id: manifest_version
+      - run: echo "The manifest is .ci/Manifest.${{ steps.manifest_version.outputs.manifest_version }}.toml"
+      - run: rm -rf .ci/Manifest.toml
+      - run: mv .ci/Manifest.${{ steps.manifest_version.outputs.manifest_version }}.toml .ci/Manifest.toml
+      - name: Cache artifacts
+        uses: actions/cache@1a9e2138d905efd099035b49d8b7a3888c653ca8 # v2.1.5
+        env:
+          cache-name: cache-artifacts
+        with:
+          path: ~/.julia/artifacts
+          key: ${{ runner.os }}-test-${{ env.cache-name }}-${{ hashFiles('**/.ci/Manifest.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-test-${{ env.cache-name }}-
+            ${{ runner.os }}-test-
+            ${{ runner.os }}-
+      - run: julia --color=yes -e 'import Pkg; Pkg.Registry.add("General")'
+        env:
+          JULIA_PKG_SERVER: ""
+      - run: julia --color=yes -e 'import Pkg; Pkg.Registry.update()'
+      - run: .ci/instantiate.sh
```

Comment:
I think maybe we should just add `RegistryCI#master` instead of using a manifest, so we always get the latest one. That way if we find a bug on staging, we can just fix it and future PRs will use the new version without a lot of manual version-bumping

Reply:
Take a look at the commit I just pushed. It deletes all the `Manifest.toml` files at the beginning. So we won't use a manifest.

Reply:
hm, that's good but we still need to add the development version or else we'll just get the release one

Reply:
ahh you do do that already, sorry!

Reply:
but actually, do we need the `run: .ci/instantiate.sh` step at all?

Reply:
It has a bounded retry loop, which is useful because the network is flaky. CI used to fail frequently just because the network is unreliable, but the CI failures stopped once we added the bounded retry loop.

Reply:
E.g. CI would fail because we download a lot of dependencies and artifacts, and sometimes one of those downloads would fail due to an intermittent network issue, and then the whole CI or AutoMerge job would fail.

Reply:
ah ok, sounds good